sparse_strips/vello_common/src/mask.rs

```diff
@@ -25,55 +29,85 @@ impl Mask {
+            usize::from(width) * usize::from(height),
+            "Expected `data` to have length of exactly `width * height`"
+        );
+
+        Self(Arc::new(MaskRepr {
+            data,
+            width,
+            height,
+        }))
+    }
+
     fn new_with(pixmap: &Pixmap, alpha_mask: bool) -> Self {
-        let data = Arc::from_iter(pixmap.data().iter().map(|pixel| {
-            if alpha_mask {
-                pixel.a
-            } else {
-                let r = f32::from(pixel.r) / 255.;
-                let g = f32::from(pixel.g) / 255.;
-                let b = f32::from(pixel.b) / 255.;
+        let data = pixmap
+            .data()
+            .iter()
+            .map(|pixel| {
+                if alpha_mask {
+                    pixel.a
+                } else {
+                    let r = f32::from(pixel.r) / 255.;
+                    let g = f32::from(pixel.g) / 255.;
+                    let b = f32::from(pixel.b) / 255.;
 
-                // See CSS Masking Module Level 1 § 7.10.1
-                // <https://www.w3.org/TR/css-masking-1/#MaskValues>
-                // and Filter Effects Module Level 1 § 9.6
-                // <https://www.w3.org/TR/filter-effects-1/#elementdef-fecolormatrix>.
-                // Note r, g and b are premultiplied by alpha.
-                let luma = r * 0.2126 + g * 0.7152 + b * 0.0722;
-                #[expect(clippy::cast_possible_truncation, reason = "This cannot overflow")]
-                {
-                    (luma * 255.0 + 0.5) as u8
+                    // See CSS Masking Module Level 1 § 7.10.1
+                    // <https://www.w3.org/TR/css-masking-1/#MaskValues>
+                    // and Filter Effects Module Level 1 § 9.6
+                    // <https://www.w3.org/TR/filter-effects-1/#elementdef-fecolormatrix>.
+                    // Note r, g and b are premultiplied by alpha.
+                    let luma = r * 0.2126 + g * 0.7152 + b * 0.0722;
+                    #[expect(clippy::cast_possible_truncation, reason = "This cannot overflow")]
+                    {
+                        (luma * 255.0 + 0.5) as u8
+                    }
                 }
-            }
-        }));
+            })
+            .collect::<Vec<u8>>();
 
-        Self {
+        Self(Arc::new(MaskRepr {
             data,
             width: pixmap.width(),
             height: pixmap.height(),
-        }
+        }))
     }
 
     /// Return the width of the mask.
+    #[inline]
     pub fn width(&self) -> u16 {
-        self.width
+        self.0.width
     }
 
     /// Return the height of the mask.
+    #[inline]
     pub fn height(&self) -> u16 {
-        self.height
+        self.0.height
     }
 
     /// Sample the value at a specific location.
     ///
     /// This function might panic or yield a wrong result if the location
     /// is out-of-bounds.
+    #[inline(always)]
     pub fn sample(&self, x: u16, y: u16) -> u8 {
         debug_assert!(
```

Comment:
```suggestion
         let repr = &*self.0;
         debug_assert!(
```

Could try dereference the arc once instead of once per access (if we haven't checked that the compiler is optimising this away).

This form might make it easier for the compiler to optimise it away completely for successive calls to `sample`